Content.Server/NPC/Systems/NPCRetaliationSystem.cs

```diff
@@ -65,25 +63,15 @@ public override void Update(float frameTime)
     {
         base.Update(frameTime);
 
-        var query = EntityQueryEnumerator<NPCRetaliationComponent, FactionExceptionComponent, MetaDataComponent>();
-        while (query.MoveNext(out var uid, out var comp, out var factionException, out var metaData))
+        var query = EntityQueryEnumerator<NPCRetaliationComponent, FactionExceptionComponent>();
+        while (query.MoveNext(out var uid, out var retaliationComponent, out var factionException))
         {
-            _deAggroQueue.Clear();
-
-            foreach (var ent in new ValueList<EntityUid>(comp.AttackMemories.Keys))
+            foreach (var entity in new ValueList<EntityUid>(retaliationComponent.AttackMemories.Keys))
             {
-                if (_timing.CurTime < comp.AttackMemories[ent])
+                if (_timing.CurTime < retaliationComponent.AttackMemories[entity])
                     continue;
 
-                if (TerminatingOrDeleted(ent, metaData))
-                    _deAggroQueue.Add(ent);
```

Comment:
You need to still remove terminating / deleted entities or else they'll throw errors when we try and access data on them. Just remove passing in the MetaData component and keep everything else the same.